[Config] Update definition.rst (#18579)
symfony/symfony-docs: components/config/definition.rst

        $databaseConfiguration,
        $configs
    );

.. caution::

    When processing the configuration tree, the processor assumes the top level 

>     array key is already stripped off. If you want a root name, you should

Reviewer: Not sure we need that second sentence as it may bring more confusion.
```suggestion
    array key (a.k.a the extension name) is already stripped off.
```

Reviewer 2: I did this change while merging. Thanks!
